### PR TITLE
Add conversion test when no API version is specified.

### PR DESCRIPTION
**Pull request by patrickdillon**
Adds a unit test for 4bea183e19d4ebaed34f1f5f0ef5a005e2ad4584. Adds further logic to conversion testing to allow for expected errors vs expected configs.

This is required because we are backporting #1760 to 4.1.z

See https://github.com/openshift/installer/pull/1764#issuecomment-505136339

### CODE REVIEW

File: pkg/types/conversion/installconfig_test.go [hunk 1]
```diff
@@ -58,6 +59,7 @@ func TestConvertInstallConfig(t *testing.T) {
 					DeprecatedClusterNetworks: []types.ClusterNetworkEntry{
 						{
 							CIDR: *ipnet.MustParseCIDR("1.2.3.5/32"),
+
```
Comment (abhinavdahiya) [inline pkg/types/conversion/installconfig_test.go]: extra line?

Comment (patrickdillon) [reply pkg/types/conversion/installconfig_test.go]: There is some trouble with gofmt here. the value should be aligned with the other values below, but the gofmt test does not like that. the extra line as seen elsewhere in the file allows this to pass that test, which seems to be incorrectly failing.

**Pull request merged**